### DISCUSSION TITLE
Customizing appbar  "toolbar" in android

### DISCUSSION THREAD

**Discussion by l-o-u-i-s-g** [beeware/toga#2616] [Q&A]
Q1:  Can the Overflow Menu  with the about  item be removed? 
I tried with: self.main_window.toolbar.clear()  but that doesn't work. 

Q2:  Is there a way to position a command  at the far left, before the app name on the toolbar?  
Typically this is where the "drawer" button would be or  a back button to return to previous screen. 

Q3:  It there a way to change the content of the label that displays the app name? 
When navigating between different screen it is useful to change this label to represent the current screen being displayed.

**Comment by freakboy3742** [comment]
Re 1. The toolbar can't currently be removed; see #1870. We have a prototype implementation of this as part of #2244. We expect to be landing this over the next month or so.

Re 2 and 3. At present, no to both. Both these essentially fall into the bigger feature request #785. This has been on my personal wish list for a long time, but I can't currently make any guarantees about when it will be implemented. What I *can* say is that #2244 should make implementing a navigation controller much easier, as that involves some low level changes to app structure that will be useful to implementing navigation.

**Reply by l-o-u-i-s-g** [reply]
1 - In this case my intentions are not to remove the  toolbar, I still want to be able to add custom commands.  I just want to be able to remove the overflow menu ( the 3 vertical dots )  and the "about".  I find this usage of the overflow rather odd.  My understanding of the overflow menu is that it is usually used when you would have too many  commands on the toolbar.  Which I guess brings in another question of if we can add other commands to this overflow menu in cases when its needed? 

PS: I'm not building any specific app yet on Toga.  I'm still experimenting with it  to understand  its functionalities and limitations. You have done a lot of good work and have high hopes for the future of this project ;)

**Reply by freakboy3742** [reply]
It uses overflow because there's no way to know how many menu items there will be; we use the "overflow" item as a catch all "root" menu from which we can hang all other menu items. This single top-level menu also supports submenus; if you put commands into Groups, each group appears as a submenu under the top level menu.

Could this be optimised? Quite possibly. I agree if you've only got 2-3 top-level menus, it might be nice to have those options displayed at the top level, rather than a single "generic" overflow menu. We're open to suggestions for other presentations of menus.

And yes, you can add other menu items - any command registered with the app should manifest as an item in the menu.

**Reply by l-o-u-i-s-g** [reply]
I would have the same 2 distinctions between menu's (commands)  and toolbar as there are on the desktop and  I would make them both optional. 

1 - I would have on the left of the label   a  drawer represented by the 3 bars icon. This would contain all the  menu items ( conmmands ) . The gmail app is a good example of this or you can see an example in flutter docs here: https://docs.flutter.dev/cookbook/design/drawer

2 - On the right of the label  I would keep the toolbar and only show an overflow if more then 3 items are added to the toolbar.

**Reply by freakboy3742** [reply]
We deliberately *don't* make a distinction between "types" of commands. In an app with good UX design, the menu contains *all* commands that are accessible to the user, and the toolbar is a subset of those commands that are useful to have a shortcut present. It isn't desirable to have a command that is *only* accessible via a toolbar; and as a result, Toga doesn't support that configuration.

As for the implementation - I agree that a side-drawer would be a useful UI element to add. However, it's less clear that it's an obvious match for hosting commands, as it is used as a *navigation* mechanism (Material3 design literally calls this widget a "NavigationDrawer"), not for exposing commands.

**Reply by l-o-u-i-s-g** [reply]
getting a bit confused with that since on the desktop app there is an extra Menu   with preferences  and quit.  Also  for example I create a group called things,  define a command  "action0" with group things  then on desktop you get a "thing" menu. with the action0 inside of if but on android I simply get  action0  directly on the toolbar. 

I agree with the side-drawer being more of a navigation mechanism, that would be perfect. But I wouldn't  duplicate its content  on the toolbar that would get too  "busy"    and   hiding a duplicate inside the overflow  would be redundant and confusing.

**Reply by freakboy3742** [reply]
Yes - the Android implementation is making a few shortcuts, and not surfacing the top level menu names. If you put a submenu *in* the "Things" menu, you it should appear as a submenu.
